setup.py

```diff
@@ -3,17 +3,25 @@
 from setuptools.command.build_ext import build_ext
 import sys
 import setuptools
+import pybind11
+import numpy as np
 
 __version__ = '0.4.0'
 
+
+include_dirs = [
+    pybind11.get_include(),
+    np.get_include(),
+]
+
 # compatibility when run in python_bindings
```

Comment:
What is this compatibility for? I think removing it should be considered. Then it would be possible to start using [cibuildwheel](https://github.com/joerick/cibuildwheel) to build and distribute wheels for linux, osx, windows.

Reply:
@groodt as @yurymalkov said there are some external scripts (outside the _nmslib/hnswlib_ repository, for example [Dockerfile.hnswlib](https://github.com/erikbern/ann-benchmarks/blob/master/install/Dockerfile.hnswlib)) that run the _setup.py_ from the _python_bindings_ folder. Therefore we added a check if _python_bindings_ in the current directory or not.

Reply:
@dyashuni @yurymalkov I'll happily update that Dockerfile. It seems to be checking out a particular branch, so we could merge or rebase master into that branch after develop is merged? It might be worth thinking about pinning to a particular version / tag / sha as well.

If this PEP-517+518 change that I've proposed is merged, then it's even easier to use hnswlib as a dependency because minimal dependencies are needed on the host, `pip install hnswlib==x.y.z` will take care of installing the build time dependencies for building the sdist.

I thik it would be even better for `hnswlib` to provide wheels in pypi because then it's even easier for to use hnswlib as a dependency because no CPP toolchains etc are necessary. This of course isn't what I'm proposing in this PR. This PR is just making this project correctly specify it's build-time dependencies so that sdist can be built reliably without manual steps for users of hnswlib.